x86/src/ops/math.rs

```diff
@@ -808,9 +913,14 @@ pub fn imul_rm8(cpu: &mut CPU, mem: Mem, instr: &Instruction) {
     cpu.regs.set16(Register::AX, res);
 }
 
-fn imul_trunc(x: i32, y: i32, _flags: &mut Flags) -> i32 {
-    // TODO: flags.
-    x.wrapping_mul(y)
+fn imul_trunc<I: SignedInt>(x: I, y: I, flags: &mut Flags) -> I {
+    let wide = x.widen() * y.widen();
+    let flag = x.checked_mul(&y).is_none();
```

Comment:
Making sure I followed: from the godbolt it looks like this doesn't multiply twice, I guess because we only check the .is_none on the result?  That is pretty cool!

Reply:
...hmm, is it possible you could have used .overflowing_mul() without needing to widen these at all?